review

tatianakharitonovich opened the pull request:
Добавлены все скрипты, добавлено модальное окно

tatianakharitonovich committed:
added styles inputRange,  separated styles global.scss

tatianakharitonovich committed:
respons

tatianakharitonovich committed:
script

tatianakharitonovich committed:
script

tatianakharitonovich committed:
script

tatianakharitonovich committed:
script

tatianakharitonovich committed:
refactoring

marinamarine commented:
общие комментарии:
1) поставь в vscode расширение ESLint, в файлах появятся подчеркивания ошибок и несоответствий стилю кода.
так же для единообразия кода поставь расширение EditorConfig for VS Code (файл .editorconfig определит общие правила для  отступов и кавычек)
так же запусти эти 2 команды из файла readme.md и посмотри ошибки в терминале http://joxi.ru/RmzBo33UjJNgam http://joxi.ru/5md1BeehaZVzJ2  
(некоторые "ошибки", например "Unexpected 'this' " внутри обработчиков событий, можно на данном этапе игнорировать).
все правила определены в файлах:
  - .stylelintrc.json 
  - .pug-lint.json
  - .eslintrc.json

2) обрати внимание на ошибки HTML-валидатора в терминале при запуске проекта http://joxi.ru/V2V63jjH8LJ7G2

3) комментарии к коммитам пишем по примеру https://docs.rs.school/#/git-convention

4) не для исправления, но для информации: в нашей компании мы используем именование модификаторов через двойной дефис (--) и разделение слов названии блока/элемента через один дефис(-)  https://ru.bem.info/methodology/naming-convention/#%D1%81%D1%82%D0%B8%D0%BB%D1%8C-two-dashes

marinamarine commented on src/pages/includes/index/blockcashbox.pug:
```diff
@@ -10,7 +10,7 @@
     .main__blockcashbox_tabsstage
         #tabcash-1
-            include aboutcash
+            include blockaboutcash
```
если название компонента/блока состоит из нескольких слов, для pug и scss разделяем слова используя _ и -
то же и для названий миксинов в pug

marinamarine commented:
- `:root` в цсс должен быть в сааамом начале, т.е. должен подключаться самым первым в `style.scss`
https://github.com/tatianakharitonovich/IRIS/blob/b3271c714550af405207ad8369ca682038215bee/src/styles/common/global.scss#L2

- компоненты - общий подход к именованию некорректен. `.main` в нашем случае блок-обертка, внутри него могут быть другие бэм-блоки со своими именами. т.е. они могут быть элементами блока .main, но чаще всего в этом нет особого смысла. поэтому каждый компонент должен иметь свое пространство имен, а внутри - не модификаторы, а элементы. например: http://joxi.ru/LmGgnvvug593Gm - т.е. основной класс называется `cards`, файл со стилями - `cards.scss`, миксин `mixin cards()` в  `cards.pug` 

- вот тут можно вдохновляться для названий классов https://github.com/metram/Common-words-of-CSS

marinamarine commented on src/styles/components/blockcashbox.scss:
```diff
@@ -0,0 +1,68 @@
+    }
+}
+
+#tabcash-2 {
```
не используем id для стилизации

marinamarine commented on src/styles/components/businesscards.scss:
```diff
@@ -0,0 +1,21 @@
+    &_title {
+      padding: 70px 0 70px 0;
+    
+      h3 {
```
у тебя в нескольких местах переопределяются стили h3 для разных элементов. тут стоило задать общий стиль для h3 в `globals.scss` и переопределять цвет или отступы через  модификатор или дополнительный класс микса https://ru.bem.info/methodology/quick-start/#%D0%BC%D0%B8%D0%BA%D1%81

в данном случае нужно консультироваться с сеошниками, можно ли использовать h3 для этих заголовков, или это должен быть див с классом например `title-h3`, т.к. сеошники очень не любят использование заголовков вне текстов

marinamarine reviewed (commented):
сравни позиционирование иконок с макетом http://joxi.ru/J2bxBjjTgLOQ7r

marinamarine commented:
- почитай про grid-gap и вообще про сетки грида, оочень плохая практика задавать высоту для блока-обертки, а вдруг в обертке будет не 4 элемента, а 6-7-8? http://joxi.ru/V2V63jjH8LJ6j2  (для чего пустые дивы???)

- вообще тут сетку вполне можно сделать через flex

- данные элементы должны быть одними и теми же на десктопе и мобиле, т.е. не должны дублироваться как у тебя, подумай, как это можно решить http://joxi.ru/eAOEWvvh6qjMam

- businesscardContent1-2-3-4 можно заменить массивом и вызывать миксин с данными через цикл

marinamarine commented on src/styles/utils/root.scss:
```diff
@@ -7,5 +7,6 @@
     --c-buttongrey: rgba(196, 196, 196, 0.8); 
     --c-formtextcolor: #898989;
-    --c-selecttextcolor: #35CC79;                             
+    --c-selecttextcolor: #35CC79;
```
разделяем слова через (-)
и постфикс color для чего если есть префикс "с"? 
пример: ` --c-select-text`

marinamarine commented on src/styles/components/businesscards.scss:
```diff
@@ -0,0 +1,21 @@
+      }
+    }
+
+    &_content {
```
хорошей практикой является общий класс `.container`, который ограничивает ширину контента каждого большого блока-компонента и центрирует (по аналогии с таким классом в bootstrap).
например для нашего макета:

```
.container {
  width: 100%; //контейнер растянется на всю доступную область
  max-width: 1280px; //но максимальная ширина 1280px - как наш макет
  padding-left: 99px; //отступы для свободных полей (на какой-то точке адаптива переопределяются на меньшее значение)
  padding-right: 99px;
  margin-left: auto; //марджины для центрирования
  margin-right: auto;
}
```

marinamarine commented on src/styles/components/businesscards.scss:
```diff
@@ -0,0 +1,21 @@
+      grid-template-columns: 48.6% 3% 48.6%;
+      grid-template-rows: 48% 4.5% 48%;
+    }
+}
```
github тут как раз намекат, что в конце каждого файла ему нужна пустая строка. это пожелание гита и хорошая практика

marinamarine commented on src/styles/style.scss:
```diff
@@ -10,12 +10,18 @@
 
-@import "components/blockorder";
\ No newline at end of file
+@import "components/blockorder";
```
вот тут наглядно видно, что делает git diff, когда нет пустой строки в конце файла :)

marinamarine commented on src/pages/includes/index/sectionscenario.pug:
```diff
@@ -35,15 +39,36 @@ include ../../components/blockscenario
+                                +scenario (scenarioContent3)
+                                +scenario (scenarioContent4)
+                    .swiper-button-next.main__blockbusinesscards_buttonright
+                        img(src='img/arrowwhite.svg', alt='')
```
кнопки слайдеров лучше стилизовать через css а не тег img
img оставим для важного контента, который потом появится в поиске по картинкам в гугле при запросе о нашем сайте, например

marinamarine commented on src/pages/includes/index/sectionscenario.pug:
```diff
@@ -35,15 +39,36 @@ include ../../components/blockscenario
+                            .main__blockscenario_slidemobile
+                                +scenario (scenarioContent4)
+                    .swiper-pagination.swiper-pagination-green                  
         #tabscen-2
```
должен быть не только id но и класс обертки

marinamarine commented on src/js/blockorder.js:
```diff
@@ -0,0 +1,380 @@
+                console.log(formArray);
+
+               
+                // fetch('https://', {
```
в финальные коммиты не должны попадать закоменченные куски кода

marinamarine commented on src/js/tabs.js:
```diff
@@ -0,0 +1,21 @@
+  nav.addEventListener('click', (event)=> showContent(event, index))
+});
+
+function showContent (event, index) {
```
объявление функции  нужно располагать раньше места вызова

marinamarine commented on src/js/tabs.js:
```diff
@@ -0,0 +1,21 @@
+  event.preventDefault();
+
+  const links = navs[index].querySelectorAll('li');
+  const tags = tabsstage[index].querySelectorAll('.main__tabsstage > div');
```
почему tags?

marinamarine commented on src/js/tabs.js:
```diff
@@ -0,0 +1,21 @@
+  links.forEach(li => li.classList.remove('tab-active'));
+  event.target.parentElement.classList.add('tab-active');
+  tags.forEach(div => {
+    div.style.display = 'none';
```
как насчет управлять свойством "display" через класс 'active' ?

tatianakharitonovich replied:
в данном случае мы меняем только 1 свойство, поэтому  управляя свойством "display" через класс код будет более громоздким.

marinamarine commented on src/pages/includes/index/blockbuy.pug:
```diff
@@ -1,41 +1,102 @@
                 br
-                select(name="buyCashbox")
-                    option(value="Нева-01-Ф") Нева-01-Ф 
+                select(name="buyCashbox", data="typeOfCash")
```
data-атрибуты немного иначе работают https://developer.mozilla.org/ru/docs/Learn/HTML/Howto/Use_data_attributes

marinamarine commented on src/styles/common/button.scss:
```diff
@@ -1,4 +1,5 @@
 @import "../utils/root";
```
не нужно импортировать рут с css переменными в разные файлы в scss, это задублирует их в билде (сейчас в style.min.css он встречается 3 раза, а должен - один и в самом начале файла)

marinamarine commented on src/styles/common/global.scss:
```diff
@@ -9,12 +10,14 @@
+  scroll-behavior: smooth;
 }
 
 body {
```
стоит так же явно указать дефолтный размер шрифта и высоту строки через относительную величину например `line-height: 1.5`

marinamarine commented on src/styles/common/global.scss:
```diff
@@ -40,20 +43,31 @@ body {
   -webkit-align-items: center;
```
префиксы для таких случаев не нужны в scss, они создаются при генерации css с помощью autoprefixer

marinamarine commented on src/styles/common/global.scss:
```diff
@@ -62,191 +76,153 @@ body {
 
-    &_tabtext {
-      padding: 0 194px 140px 194px;
+  &__tabsstage > div:first-child{
```
предпочтительно вложенным дивам дать классы

tatianakharitonovich replied:
все вложенные дивы имеют классы, но так как табы у нас в проекте встречаются несколько раз, то классы у вложенных дивах разные, данное обращение в нашем случае универсальное для всех табов

marinamarine replied:
у этих дивов может быть так же класс, определяющий, что это контент таба с какими-то общими стилями (например, отступами)

marinamarine commented on src/styles/components/blockaboutcash.scss:
```diff
@@ -3,33 +3,93 @@
+    &_abouttitle, &_abouttitlemobile {
         padding-bottom: 53px;
 
         h4 {
```
то же что и про h3

tatianakharitonovich committed:
refactor: make changes to comments

fix HTML validator errors,
rename block section ckasses,
rename mixins, blocks,
add classes to blocks with id,
replace headers h3-h4 with containers with classes title-h3, title-h4,
change block grid
remove duplicate busnesscards
rename colors,
add empty strings,
change the styling of the slider buttons,
set default font size and line height
remove prefixes,
move function declarations above the call site

marinamarine commented:
немного по визуалу:
- цвет основного фона не по макету (плюс на адаптиве цвет должен измениться), лого немного слишком уехало вверх, положение меню, отступ до первого заголовка http://joxi.net/823WvVVHajN9X2
- у фиксированного хэдера тень http://joxi.ru/zANM6vvh8zpvL2
- на "коротких" экранах нельзя добраться до кнопки в попапе http://joxi.ru/4AkLljjT0EPkpr
- черная рамка вокруг одного элемента предполагает, что весь блок интерактивный, а не только стрелочка http://joxi.ru/BA0QGVVHv7yPQr

marinamarine commented on src/pages/components/section.pug:
```diff
@@ -1,10 +1,13 @@
\ No newline at end of file
+      .section__picture
+        img(alt='schedule', src=content.src, width=content.width)
+      .section__picturemobile
```
для таких случаев лучше использовать тег picture с атрибутом media в source

tatianakharitonovich committed:
refactor: make changes to visual

add shadow for header
change logo position in header
change navigation position in header
change body color
replace tag img to picture
add classes to div in tabs
change form size in block-order
add interactivity to business-cards

tatianakharitonovich committed:
refactor
